lib/Frontend/CompilerInvocation.cpp

```diff
@@ -47,7 +47,25 @@ static void updateRuntimeLibraryPath(SearchPathOptions &SearchPathOpts,
   llvm::sys::path::append(LibPath, getPlatformNameForTriple(Triple));
   SearchPathOpts.RuntimeLibraryPath = LibPath.str();
 
-  llvm::sys::path::append(LibPath, Triple.getArchName());
+  // The linux provided triple for ARM contains a trailing 'l'
+  // denoting little-endian.  This is not used in the path for
+  // libraries.  LLVM matches these SubArchTypes to the generic
+  // ARMSubArch_v7 (for example) type.  If that is the case,
+  // use the base of the architecture type in the library path.
+  if (Triple.isOSLinux()) {
```

Comment:
Because the fact that there are still many showstopper issues with big endian (which were brought up in the discussion of the c-based implementation of the conformance objects) it may not make sense to put the cart before the horse on this one.  I doubt this would pose undue complication to a future effort supporting big-endian arm cores.